src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -336,16 +336,27 @@ internal void EnsureTemplate()
 		}
 
 		/// <summary>
-		/// Finds a realized element in the control template
+		/// Finds a realized element in the control template.
 		/// </summary>
-		/// <param name="e">The framework element instance</param>
-		/// <param name="name">The name of the template part</param>
+		/// <param name="childName">The name of the template part.</param>
+		/// <returns>The first template part of the specified name; otherwise, null.</returns>
 		public DependencyObject GetTemplateChild(string childName)
 		{
 			return FindNameInScope(TemplatedRoot as IFrameworkElement, childName) as DependencyObject
 				?? FindName(childName) as DependencyObject;
 		}
 
+		/// <summary>
+		/// Finds a realized element in the control template of the specified type.
+		/// </summary>
+		/// <typeparam name="T">The type of the template part.</typeparam>
+		/// <param name="childName">The name of the template part.</param>
+		/// <returns>The first template part of the specified name; otherwise, null.</returns>
+		public T GetTemplateChild<T>(string childName) where T : class, DependencyObject
```

Comment:
This is a nice idea, but it should be an internal method. We want to avoid adding stuff that isn't part of the UWP contract, even helpful stuff like this, because it allows people to write code that compiles on Uno targets but not UWP.

```suggestion
		internal T GetTemplateChild<T>(string childName) where T : class, DependencyObject
```

Reply:
Yes, you're absolutely right I missed that when it was copy-pasted from my private code base.